### DISCUSSION TITLE
How can I save the pictures in the webpage? And...

### DISCUSSION THREAD

**Discussion by ajmz** [coddingtonbear/obsidian-web#7] [Q&A] [answered]
1. How can I save the pictures in the webpage?
2. How to save the whole page?
3. It is suggested that it can choose to save the directory in the interface.
4. It is recommended to refer to ”Joplin Web Clipper“

![2022-03-19_033751](https://user-images.githubusercontent.com/30185045/159073119-93e8cab3-ea68-4adb-8f32-4e810d164390.png)

**Comment by coddingtonbear** [comment, marked as answer]
I transferred this from the Obsidian Local REST API project because I believe you assumed that the features you've asked about might be provided by the REST API.  They're not; they're instead provided by the Obsidian Web chrome plugin.

I'll try to address these one-by-one:

1. *How can I save the pictures in the webpage?* -- This plugin doesn't provide that feature.
2. *How to save the whole page?* -- Look at the "Capture Page Snapshot" template in Obsidian Web for an example.  The page content is always available in your templates in the `{{ page.content }}` context variable.
3. *It is suggested that it can choose to save the directory in the interface* -- I'm not exactly sure how to parse this question, but I'll interpret it as "How do I select the directory to save the content to?".  If you look at any of the existing templates, you'll see that they're interacting with the https://github.com/coddingtonbear/obsidian-local-rest-api plugin in Obsidian.  You can configure the operation used for interacting with the API including the path (directory) that the file will be stored in by adjusting the URL.  See https://coddingtonbear.github.io/obsidian-local-rest-api/ for more information.
4. *It is recommended to refer to "Joplin Web Clipper"*  I'm at a little bit of a loss trying to understand what exactly this question might be, but that does look neat.  Unfortunately it's for a different system, though; so I'm not super sure what you're trying to communicate.

Hope this helps!

**Comment by coddingtonbear** [comment]
If somebody would like to take the time to add support for downloading images for a webpage (item 1 in the above list), I'm interested in helping talk through what a pull request for such a feature might look like, but I'm afraid I don't have the time to add the feature myself.

**Reply by ajmz** [reply]
> If somebody would like to take the time to add support for downloading images for a webpage (item 1 in the above list), I'm interested in helping talk through what a pull request for such a feature might look like, but I'm afraid I don't have the time to add the feature myself.

I am not a programmer, but by searching for keywords, I found the following source code page about  Joplin Web Clipper, which may be related to the browser plug-in save images, for reference only. BTW, Joplin is MIT protocol.

https://github.com/laurent22/joplin/blob/46868e6b25e3c5ae2360a1a009fe7dd4866870c3/packages/app-clipper/content_scripts/index.js

**Reply by coddingtonbear** [reply]
Thanks for the code reference, @ajmz.  Unfortunately, although we all wish it were simpler, code interacting with different systems is far from fully portable; so although reading their source might help make it easier to get an understanding of the full scope of the problem being solved, it'll still require a significant amount of effort from somebody willing to do a little programming to make the feature usable here.

You're welcome to give it a shot, though -- I'd be glad to provide a little guidance on how you might approach replicating the behavior of that plugin here.

**Reply by ajmz** [reply]
It's too hard for me as a non-programmer. 
I found another plugin for Obsidian, Obsidian Local Images.
“The plugin finds all links to external images in your notes, downloads and saves images locally and finally adjusts the link in your note to point to the local image files.”
https://github.com/aleksey-rezvov/obsidian-local-images
Maybe you can combine it with your project. 
Otherwise, send this idea to obsidian's official forum to find participants. I saw your previous posts before.

**Reply by coddingtonbear** [reply]
Unfortunately, I have way too much going on to field that kind of conversation, but you're welcome to post something there and try to put together a team toward that end if you'd like.

**Reply by ajmz** [reply]
Could you please add an option in Local REST API for Obsidian or obsidian-Web, that automatically executes a set of user-configurable shortcuts after a web page is saved?
Because Obsidian-local-images has the shortcut: Download images locally -- your active page will be processed.
Of course, it can also give users the option to automatically handle other operations after the web page is saved.

**Reply by coddingtonbear** [reply]
You should really file that as a separate discussion -- it's only loosely related to this discussion, but if your question is around whether _I_ would add that: the answer is that it's extremely unlikely I'll be personally adding features unless they happen to line up very closely with pain points I personally experience.  I have extremely limited time and a lot of other things to juggle; so in this case the answer is, unfortunately, a very solid "no".  In this case, like others, you're going to need to either write the solution yourself or find somebody else to make a feature proposal & write the feature.
